doc: nrf91: snippet page edits (#16329)
nrfconnect/sdk-nrf: doc/nrf/device_guides/nrf91/nrf91_snippet.rst

   :local:
   :depth: 2

nRF91 Series' :ref:`app_build_snippets` are tailored for tracing on the nRF91 Series devices but can work with other boards as well.

> On nRF91 Series devices, the following functionalities can be enabled using snippets:

Reviewer: ```suggestion
On nRF91 Series devices, you can enable the following functionalities using snippets:
```

.. list-table::
   :header-rows: 1

   * - Functionality
     - Snippet name
     - Compatible board targets
   * - :ref:`nrf91_modem_trace_ext_flash_snippet`
     - ``nrf91-modem-trace-ext-flash``
     - ``nrf9151dk/nrf9151/ns``, ``nrf9161dk/nrf9161/ns``, ``nrf9160dk/nrf9160/ns``, ``nrf9131ek/nrf9131/ns``
   * - :ref:`nrf91_modem_trace_uart_snippet`
     - ``nrf91-modem-trace-uart``

>      - ``nrf9151dk/nrf9151/ns``, ``nrf9161dk/nrf9161/ns``, ``nrf9160dk/nrf9160/ns``, ``nrf9131ek/nrf9131/ns``

Reviewer 2: ```suggestion
     -  :ref:`All nRF91 Series board targets <ug_nrf91>`
```

   * - :ref:`tfm_enable_share_uart`
     - ``tfm-enable-share-uart``

>      - :ref:`All nRF91 Series board targets <ug_nrf91>`

Author: @eivindj-nordic , can you confirm lines 21, 24, and 27 in this table? Thanks.

The snippet also enables the UART1 peripheral with a baud rate of 1 Mbd and hardware flow control enabled.
If this configuration does not match your requirements, you can add a snippet or Kconfig and devicetree overlays to your application with the desired setup.

.. note::
    If you are using the nRF9160 DK, remember to :ref:`set the board controller switch to the **nRF91** position <build_pgm_nrf9160_board_controller>` before programming.

To enable modem tracing with the UART trace backend on a nRF91 device, add the ``nrf91-modem-trace-uart`` snippet to the :term:`build configuration`.
This can be done in one of the following ways:

.. tabs::

   .. group-tab:: west

      To add the modem trace UART snippet when building an application with west, use the following command pattern, where *board_target* corresponds to your development kit board target and `<image_name>` to your application image name:

      .. parsed-literal::
        :class: highlight

        west build --board *board_target* -- -D<image_name>_SNIPPET="nrf91-modem-trace-uart"

      .. note::

>         With :ref:`sysbuild <configuration_system_overview_sysbuild>`, using the ``west build -S`` option applies the snippet to all images.
>         Therefore, use the CMake argument instead, specifying the application image.

Reviewer: ```suggestion
         With :ref:`sysbuild <configuration_system_overview_sysbuild>`, using the ``west build -S`` option applies the snippet to all images.
         Therefore, use the CMake argument instead, for specifying the application image.
```

Reviewer: I'm not sure whether I understand this second sentence. Use CMake argument instead of what? Also the subclause after the comma is also a bit odd. Would it need something more, like when specifying or for specifying.

   .. group-tab:: CMake

      To add the modem trace UART snippet when building an application with CMake, add the following command to the CMake arguments:

      .. code-block:: console

        -D<image_name>_SNIPPET="nrf91-modem-trace-uart" [...]

      To build with the |nRFVSC|, specify ``-D<image_name>_SNIPPET="nrf91-modem-trace-uart" [...]`` in the **Extra CMake arguments** field.

      See :ref:`cmake_options` for more details.

.. _tfm_enable_share_uart:

Shared UART for application and TF-M logging
********************************************

If you want to activate TF-M logging while having modem traces enabled, it can be useful to direct the TF-M logs to the UART (**UART0**) used by the application.

> To activate both modem traces and TF-M logs, use the following command pattern, where *board_target* corresponds to your development kit board target:

Reviewer 2: ```suggestion
To activate both modem traces and TF-M logs, use the following command pattern, where *board_target* corresponds to your board target:
```
